### DISCUSSION TITLE
Keycloak 23: Script upload is disabled while trying to import authorization settings

### DISCUSSION THREAD

**Discussion by vanceism7** [keycloak/keycloak#28019] [Q&A]
Hello, as the title mentions, I've run into the issue that I'm not able to import authorization settings. 

The basic scenario is that I've installed keycloak into a kubernetes cluster using bitnami's helm keycloak chart. So far everything has basically been fine, but as I was learning how to setup authorization permissions, I created a test realm to try things out in. After getting a gist of what I was doing, I learned I could export my authorization settings and import them back into another realm. 

When I try to import the auth settings into another realm, I simply get back "An unknown error has occurred" in the UI. Upon looking at the logs for my keycloak servers in the backend, I see the error "Script upload is disabled". 

I've done a lot of internet searching on this, but it looks like the majority of issues people have had has been regarding importing an entire realm into keycloak, which is not the same scenario I'm dealing with. Additionally, it seems like a lot of this was regarding people migrating up to a newer version and having old script references left behind - this is also not the case for me as this is a fresh installation of v23. 

Any help to get the authz settings upload working would be greatly appreciated!
Thanks!

**Comment by vanceism7** [comment]
Oh blah nevermind! Turns out I did have a default js policy which was setting off the "script upload" alarm.

**Reply by SurajWijewickrama-acs** [reply]
I have the same issue, can you tell me how you fixed it? where is the default js policy?

**Reply by vanceism7** [reply]
When you create a new client for your realm (assuming you've enabled authorization), you'll find the Default Policy under `Authorization` > `Policies`. 

![image](https://github.com/keycloak/keycloak/assets/2016375/0431a937-8af6-4e0d-a696-6bb1164b1a30)

You have to delete that `Default Policy` prior to exporting your authz settings. Otherwise, when you try to import that authz config, the import will fail because `Default Policy` is a js script, which is disallowed without the `import script` functionality turned on

**Reply by sachingaikwad123** [reply]
This is very inconvenient.

Can "export" feature have toggle option to "exclude "Authorization Policy Settings" (JS script based)? This will help export and import easily.

**Reply by Mario-Coxe** [reply]
This solved my issue. Removing the Default Policy before export fixed the import error. Thanks!

**Comment by fareskhm** [comment]
I'm encountering the same issue with Keycloak version 26.4.11 (Red Hat Build of Keycloak). When creating a new client with Authorization enabled, Keycloak automatically generates a "Default Policy" of type JavaScript under Authorization → Policies. This default JS policy causes the realm export/import pipeline to fail with the error Script upload is disabled
